### PR TITLE
Lesson 1 java

### PR DESCRIPTION
**Pull request by davidabrahams**

### CODE REVIEW

File: lesson 1/README.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+}
+```
+
+A window button should appear next to the main method header. Click it to run your program!
+
+```
+Hello World!
+```
+
+Don't worry too much yet about what all the words in `public static void main(String[] args)` mean. For now, just remember that this is the **main method**, and that every java program must have a main method defined. This main method is what the Java Virtual Machine runs initially, and all subsequent code eventually stems from the main method.
```
Comment (shrutiyer) [inline lesson 1/README.md:44]: Do we ever explain String[] args or about void? I know we talk about public, static, etc

File: lesson 1/README.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+}
+
+```
+
+##### Constructors
+
+The constructor is a special method used to initialize instances of your class (just like Python's `__init__` method). Classes by default inherit a no-argument constructor from the default `Object` class. That means that even without epxlicitly defining a constructor, we can already initalize an `Account`:
+
+```Account a = new Account() // This works!```
+
```
Comment (shrutiyer) [inline lesson 1/README.md:114]: Also, I remember that you could have multiple constructors right? Each one with different arguments

File: lesson 1/README.md [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+Often, you will want to initialize your object with some values. In this case you must define your own constructor.
+
+**Go ahead and implement the `Account` constructor. It should have input arguments `long amount, MoneySaver owner`**
+
+#### Fields
+
+You'll notice both classes also have fields. Fields are either attributes of members of a class (an `amount` describes an `Account`) or things that belong to the class (an `Account` has an `owner`). **Fields should almost exclusively be `private` unless you have a *really* good reason to make it `public`.** If you want an outside class to be able to access a field, you must create public getter/setter methods.
```
Comment (shrutiyer) [inline lesson 1/README.md]: We should talk about protected too

File: lesson 1/README.md [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+
+```
+Account owner: Jim, Account Balance: $100
+New amount: 20
+```
+
+#### Methods
+
+These classes also have methods, which are things instances can "do". Each method must specify what it returns, (`void` if nothing). Generally speaking methods are `public`. However, you might find yourself writing a `private` method that nothing outside the class should be able to see. Often, "helper" methods are `private`.
+
```
Comment (shrutiyer) [inline lesson 1/README.md:145]: Mention getters and setters? And how the variable should be private but getters and setters public

File: lesson 1/README.md [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+```
+Account owner: Jim, Account Balance: $100
+New amount: 30
+```
+
+##### What does `static` mean?
+
+`static` methods operate on the *Class*, as opposed to *instances* of the class. You can call static methods without creating any object e.g. `Collections.sort()`. Make a method `static` if the method is something the *Class* can do.
+
```
Comment (shrutiyer) [inline lesson 1/README.md:167]: something the 'Class' can do **and doesn't need any instance variables to work properly**

**Comment by nmohamed** [comment]
how do you plan on walking through this lesson? it seems like it's just "look at the readme.md and follow what it says" which is fine if we want to do that

**Comment by davidabrahams** [comment]
@nmohamed I think I was going to briefly introduce them to classes, fields, and methods, and then set them loose.

File: lesson 1/README.md (hunk 1 above)
Comment (davidabrahams) [inline lesson 1/README.md:44]: I honestly don't even know what it is. I assume its command line arguments you give it? It's not really worth explaining.

File: lesson 1/README.md (hunk 2 above)
Comment (davidabrahams) [inline lesson 1/README.md:114]: added a line about this

File: lesson 1/README.md (hunk 3 above)
Comment (davidabrahams) [inline lesson 1/README.md]: really? I didn't learn about protected for a while when first learning Java....You can get away with only doing private and public when you start out.

File: lesson 1/README.md (hunk 4 above)
Comment (davidabrahams) [inline lesson 1/README.md:145]: I did mention getters and setters in the fields section

**Comment by billmwong** [comment]
Review: 👍

**Pull request merged**